TutorLizard.BusinessLogic/Services/StudentService.cs

```diff
@@ -17,26 +17,70 @@ public AdRequest CreateAdRequest(int adId, string message)
     {
         // return created request
         // return empty with id = 0 to indicate something went wrong
+        int? studentId = _userIdentityService.GetUserId();
+        
+        if (studentId is null) 
+        {
+            return new AdRequest()
+            { AdId = 0 };
+        }
+
+        return _dataAccess.CreateAdRequest(adId, studentId.Value, isAccepted: false, message);
         throw new NotImplementedException();
     }
     public ScheduleItemRequest CreateScheduleItemRequest(int scheduleItemId)
     {
         // return created request
         // return empty with id = 0 to indicate something went wrong
+        int? studentId = _userIdentityService.GetUserId();
+
+        if (studentId is null)
+        {
+            return new ScheduleItemRequest
+            {
+                ScheduleItemId = 0
+            };
+        }
+        return _dataAccess.CreateScheduleItemRequest(scheduleItemId, studentId.Value);
         throw new NotImplementedException();
     }
     public List<Ad> GetAllAds()
     {
+        if (_dataAccess.GetAllAds() is IEnumerable<Ad> allAds) 
+        {
+            List<Ad> ads = allAds.ToList();
+            return ads;
+        }
         throw new NotImplementedException();
     }
     public List<Ad> GetUsersAcceptedAds()
     {
         // list of ads that the active user created request for and they were accepted
+        int userId = (int)_userIdentityService.GetUserId();
```

Comment:
Don't cast it to int before you check it's not null. It could cause an exception.